discord/state.py

```diff
@@ -661,41 +693,44 @@ def parse_presence_update(self, data):
 
         self.dispatch('presence_update', old_member, member)
 
-    def parse_user_update(self, data):
-        self.user._update(data)
-        ref = self._users.get(self.user.id)
+    def parse_user_update(self, data) -> None:
+        # self.user is *always* cached when this is called
+        user: ClientUser = self.user # type: ignore
+        user._update(data)
+        ref = self._users.get(user.id)
         if ref:
             ref._update(data)
 
-    def parse_invite_create(self, data):
+    def parse_invite_create(self, data) -> None:
         invite = Invite.from_gateway(state=self, data=data)
         self.dispatch('invite_create', invite)
 
-    def parse_invite_delete(self, data):
+    def parse_invite_delete(self, data) -> None:
         invite = Invite.from_gateway(state=self, data=data)
         self.dispatch('invite_delete', invite)
 
-    def parse_channel_delete(self, data):
-        guild = self._get_guild(utils._get_as_snowflake(data, 'guild_id'))
+    def parse_channel_delete(self, data) -> None:
+        guild = self._get_guild(utils._get_as_snowflake(data, 'guild_id')) # type: ignore
         channel_id = int(data['id'])
         if guild is not None:
             channel = guild.get_channel(channel_id)
             if channel is not None:
                 guild._remove_channel(channel)
                 self.dispatch('guild_channel_delete', channel)
 
-    def parse_channel_update(self, data):
+    def parse_channel_update(self, data) -> None:
         channel_type = try_enum(ChannelType, data.get('type'))
         channel_id = int(data['id'])
         if channel_type is ChannelType.group:
             channel = self._get_private_channel(channel_id)
             old_channel = copy.copy(channel)
-            channel._update_group(data)
+            # the channel is a GroupChannel
+            channel._update_group(data) # type: ignore
             self.dispatch('private_channel_update', old_channel, channel)
             return
 
         guild_id = utils._get_as_snowflake(data, 'guild_id')
-        guild = self._get_guild(guild_id)
+        guild = self._get_guild(guild_id) # type: ignore
```

Comment:
I assume this was because `guild_id` is `Optional[int]`. The signature should be updated instead.

Reply:
Do you mean typehinting `guild_id` as `int` and `# type: ignore`'ing that line, or changing the signature of `_get_as_snowflake`? I don't see how `_get_as_snowflake`'s signature could be changed if that was the case.

Reply:
I'm talking about `_get_guild`'s signature.

Reply:
Alright. Should the signature for all `ConnectionState._get_x` methods that take an id be changed for the id to be `Optional[int]`?

Reply:
Sure.